charts/legend-deployment/templates/external-secret.yaml

```diff
@@ -0,0 +1,19 @@
+{{- range $secret := .Values.deployment.container.externalSecrets}}
+apiVersion: external-secrets.io/v1beta1
+kind: ExternalSecret
+metadata:
+  name: {{ $secret.name }}
+spec:
+  refreshInterval: 1h                            # rate SecretManager pulls GCPSM
+  secretStoreRef:
+    kind: SecretStore
+    name: google-secret-manager                  # name of the SecretStore (or kind specified)
+  target:
+    name: {{ $secret.name }}              # name of the k8s Secret to be created
+    creationPolicy: Owner
+  data:
+    - secretKey: {{ $secret.secretKey }}  # name of the GCPSM secret key
```

Comment:
Seems like only support one key-value pair per secret with this implementation. 
Have we also considered supporting secrets like this: 
![image](https://github.com/user-attachments/assets/3831b5b7-118e-480e-9c96-6d350caa670f)